p2p/src/net/libp2p/discovery/mdns.rs

```diff
@@ -96,7 +102,7 @@ mod tests {
 
     #[tokio::test]
     async fn mdns_disabled() {
-        let mdns = Mdns::new(false).await;
+        let mdns = Mdns::new(false, 0).await;
```

Comment:
What does 0 mean? If I understand correctly, 0 has no effect because mdns is disabled. I would rather then see an `Option<Duration>` or `Option<Settings>` instead of values with no effect.

Reply:
Btw, a separate enum would be better than Option.

Reply:
0 is what `Default::default()` gives for `std::time::Duration` which is no timeout. I want to preserve the default behavior of libp2p's mDNS so I need to think about this.

Reply:
It now accepts an `Option` for the query interval but I'm not sure if it's any better if I'm honest